component/registration_form.jsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'preact/hooks'
+import { Form, Text } from './form.jsx'
+
+export const RegistrationForm = () => {
+  let [errors, setErrors] = useState({})
+
+  const onSubmit = async (e) => {
+    e.preventDefault()
+    // form data
+    const form = new FormData(e.target)
+    const data = Object.fromEntries(form)
+    const json = JSON.stringify(data)
+    console.log('json', json)
+    const fetching = await fetch('/api/user/registerForm', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      credentials: 'include',
+      body: json,
+    })
+    const response = await fetching.json()
+    if (response?.errors) {
+      setErrors(response.errors)
+    } else {
+      console.log(response)
+      // redirect here
+    }
+  }
+
+  return (
+    <>
+      <Form onSubmit={(e) => onSubmit(e)} title="Profile-Info" submit="save">
+        <PhoneInput
+          name="ownContact"
+          comment="please enter your phone number"
+          errors={errors}
+        />
+        {'\n'}
+        <PhoneInput
+          name="emergencyContact"
+          comment="please enter a phone number to join in case of emergency"
+          errors={errors}
+        />{' '}
+      </Form>
+    </>
+  )
+}
+
+const check = ({ target }) => {
+  let formatRule = target.value.includes(' ')
+    ? target.value.split(' ').join('')
+    : target.value
+  let prefix = formatRule.slice(0, 4)
+  let rest = formatRule.split('').slice(4)
+  let count = 1
+  rest = rest.flatMap((value, index) => {
+    if (index < 3) {
+      return (index > 1 && index % 2 === 0) || index === 0
+        ? [' ', value]
+        : [value]
+    } else {
+      count += 1
+      if (count === 3) {
+        count = 0
+        return [' ', value]
+      } else {
+        return [value]
+      }
+    }
+  })
+  target.value = prefix + rest.join('')
+  target.setSelectionRange(target.value.length, target.value.length)
```

Comment:
add the formatted rest array (transform in string) to prefix for have full number : +225 01 01 722 201 , and put the cursor to the end of the input